### PR TITLE
Add COMPASS test cases with Ice Shelf Cavities

### PR DESCRIPTION
**Pull request by mark-petersen**
This PR adds three new cases to COMPASS:
- QU240wISC
- EC60to30wISC
- SO60to10wISC

These are all 80-layer cases. 

They each include the following steps:
- init
  - base_mesh
  - culled_mesh
  - initial_state
  - ssh_adjustment
  - test
- spin_up
  - prep_spin_up1
  - simulation

All cases have been created and spun up successfully.

### CODE REVIEW

**Comment by mark-petersen** [comment]
These all appear to be working. I'm testing spin-up now. Here are mesh images:

**QU240ISC**
image

**Comment by mark-petersen** [comment]
**EC60to30**
image
path on LANL IC:
/lustre/scratch4/turquoise/mpeterse/runs/190715a_EC60to30ISC/ocean/global_ocean/EC60to30ISC

**Comment by mark-petersen** [comment]
**SO60to10ISC**
image
Path on LANL IC:
/lustre/scratch4/turquoise/mpeterse/runs/190715a_SO60to10ISC/ocean/global_ocean/SO60to10ISC/init/test

**Comment by mark-petersen** [comment]
**SO60to10ISC** zoom, note how strongly layer thickness varies on the right.
image

File: testing_and_setup/compass/ocean/global_ocean/SO60to10ISC/init/define_base_mesh.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+def cellWidthVsLatLon():
+    lat = np.arange(-90, 90.01, 0.1)
+    lon = np.arange(-180, 180.01, 10.0)
+
+    cellWidth10 = 10.0 * np.ones(lat.size)
+    cellWidth30 = 30.0 * np.ones(lat.size)
+
+    latTransition = -40.0
+    latWidthTransition = 5.0
```
Comment (mark-petersen) [inline testing_and_setup/compass/ocean/global_ocean/SO60to10ISC/init/define_base_mesh.py]: I decided to only add a single case for SO60to10ISC. The specification of the location and width of the southern transition is here, so it is easy enough to alter.

**Comment by mark-petersen** [comment]
@kristin-hoch could you test these three cases from scratch on LANL IC, make sure they work for you, and post any comments?  For reference, here is the normal workflow:
```
git clone git@github.com:mark-petersen/MPAS-Model.git compass_EC60to30ISC_QU240ISC
cd !$
git checkout -b compass_EC60to30ISC_QU240ISC origin/ocean/compass_EC60to30ISC_QU240ISC
```
Compile with your typical modules. Then
```
module unload python
source /usr/projects/climate/SHARED_CLIMATE/anaconda_envs/load_compass.sh
cd testing_and_setup/compass/
```
Please test all of these:
```
./list_testcases.py |grep ISC
  49: -o ocean -c global_ocean -r EC60to30ISC -t init
  50: -o ocean -c global_ocean -r EC60to30ISC -t spin_up
  62: -o ocean -c global_ocean -r QU240ISC -t init
  65: -o ocean -c global_ocean -r SO60to10ISC -t init
  66: -o ocean -c global_ocean -r SO60to10ISC -t spin_up
```
for example:
```
./setup_testcase.py -n 49 -f general.config.ocean_turq --work_dir /lustre/scratch4/turquoise/mpeterse/runs/190716_test_compass_ISC_cases
```
with your path. The QU240 can run on one node; all base_mesh and cull_mesh are on one node. EC and SO mpiruns are set up for 4 nodes. Everything should work with no namelist alterations. Thanks for your help!

**Comment by mark-petersen** [comment]
**EC60to30ISC** 
Kinetic Energy after 20 days of simulation (after 10-day spin-up step), layer 1 and 20 (150m depth), on log color scale. Note circular waves show SSH is still adjusting. Agulhas current is already there, and Antarctic coastal currents starting. 
image

File: testing_and_setup/compass/ocean/global_ocean/SO60to10ISC/init/define_base_mesh.py (hunk 1 above)
Comment (mark-petersen) [reply testing_and_setup/compass/ocean/global_ocean/SO60to10ISC/init/define_base_mesh.py]: image

File: testing_and_setup/compass/ocean/global_ocean/config_files_ISC/config_initial_state.xml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	<streams name="streams.ocean" keep="immutable" mode="init">
+		<template file="template_initial_state.xml" path_base="script_configuration_dir"/>
+		<template file="template_critical_passages.xml" path_base="script_core_dir" path="global_ocean"/>
+	</streams>
+
+	<run_script name="run.py">
+		<step executable="gpmetis">
+			<argument flag="graph.info">4</argument>
+		</step>
+		<model_run procs="4" threads="1" namelist="namelist.ocean" streams="streams.ocean"/> </run_script> </config>
```
Comment (xylar) [inline testing_and_setup/compass/ocean/global_ocean/config_files_ISC/config_initial_state.xml]: Strange formatting here.  
```suggestion
		<model_run procs="4" threads="1" namelist="namelist.ocean" streams="streams.ocean"/> 
	</run_script>
</config>
```

**Comment by xylar** [comment]
> SO60to10ISC zoom, note how strongly layer thickness varies on the right.

I'm not sure if your comment was because you're worried about that, but it's expected.  It's the only way to prevent the Haney number from going over 5.

File: testing_and_setup/compass/ocean/global_ocean/config_files_ISC/config_initial_state.xml (hunk 2 above)
Comment (mark-petersen) [reply testing_and_setup/compass/ocean/global_ocean/config_files_ISC/config_initial_state.xml]: fixed, thanks.

**Comment by mark-petersen** [comment]
> >SO60to10ISC zoom, note how strongly layer thickness varies on the right.
>
>I'm not sure if your comment was because you're worried about that, but it's expected. It's the only way to prevent the Haney number from going over 5.

Not particularly worried, just making a note. If we have numerical instabilities that is the most likely place, as we already know.

**Comment by mark-petersen** [comment]
Does anyone have a preference between ISC and Cav or something else for 'with Ice Shelf Cavities'?
EC60to30ISC versus EC60to30Cav
They both seem non-intuitive. Maybe ISC is fine for a simple tag to differentiate.

**Comment by xylar** [comment]
A slight variant would be `wISC`, which has the advantage of being pronounceable and whimsical, "wisk!"

**Comment by mark-petersen** [comment]
Yes, that might be better. I keep looking at EC60to30ISC and my brain turns 0ISC into Disk, which is senseless.  Let's see:
- QU240wISC
- EC60to30wISC
- SO60to10wISC

**Comment by mark-petersen** [comment]
@xylar great news! The SO60to10wISC spun up successfully, with:
**Stage 1:   5 days**
```
config_dt = '00:05:00'
config_btr_dt = '00:00:15'
config_Rayleigh_friction = .true.
config_Rayleigh_damping_coeff = 1.0e-4
```
**Stage 2:   3 days**
```
config_dt = '00:07:30'
config_btr_dt = '00:00:15'
config_Rayleigh_friction = .false.
```
I'm was hoping to run with a 10 minute timestep, but it usually fails. I'll see if I can get it higher.

Here is Kinetic Energy, layer 1 and 20, at 8 days. Everything appears fine. The Southern Ocean has some background barotropic motion, but maybe it's still spin-up.
image

I think I got over the spin-up hump where you were stuck previously. (See [previous Southern Ocean design page](https://acme-climate.atlassian.net/wiki/spaces/ECG/pages/969211983/Design+of+Southern+Ocean+Enhanced+Mesh#DesignofSouthernOceanEnhancedMesh-ANEC60to30cr5,transitionasdistancefromcoastline)). Yours was 5km below ice shelves, mine is 10km, so maybe that's the difference. I could go make another with 5km and just see if it works.

**Comment by xylar** [comment]
@mark-petersen, is there a reason this got closed?

**Comment by mark-petersen** [comment]
Whoops! I forgot I made this PR and had renamed the branch with a more precise name. When you delete a branch on the fork, it closes it automatically. @xylar, thanks for asking.

**Comment by mark-petersen** [comment]
@xylar @stephenprice @kristin-hoch:
FYI, Kristin will be helping with the creation, discussion, and spin-up of the two meshes we discussed at [last Wednesday's meeting.](https://acme-climate.atlassian.net/wiki/spaces/ECG/pages/1025900731/2019-08-26+Special+Topics+var.+res.+ocean+ice+configuration+for+v2+cryo+simulations) 

1. SO60to10ISC, see [plot of grid cells vs latitude](https://user-images.githubusercontent.com/4036816/61309398-61985780-a7af-11e9-9a7d-32175e4669e3.png)
2. Same but with refined 5km around Antarctica.

Kristin, use this branch to get started. If you make local changes, we can add them to this branch once we've settled on them.

**Comment by xylar** [comment]
@mark-petersen, I'd like to have  #381 addressed in this branch if you're okay with that.  Would you like to make the edit or shall I?

**Comment by mark-petersen** [comment]
Rebased and retested.
@xylar check that last commit. Fixes https://github.com/MPAS-Dev/MPAS-Model/issues/381.
This PR will get merged late this week or next week.

**Comment by xylar** [comment]
Looks good.  I'll rebase what I'm working on and test tomorrow.

**Comment by kristin-hoch** [comment]
I've added changes for the Southern ocean case with the transition at 48S to 10 km and a widened transition region.

**Comment by mark-petersen** [comment]
Rebased and tested on IC. This PR successfully creates and spins up these cases.

**Pull request merged**